HACKING-URL.md

```diff
@@ -0,0 +1,319 @@
+- [URL editing handlers](#url-editing-handlers)
+  * [Cleanup](#cleanup)
+  * [Autoselection](#autoselection)
+  * [Validation](#validation)
+  * [Supporting new relationship types](#supporting-new-relationship-types)
+  * [Entity-wide rules](#entity-wide-rules)
+- [URL display handlers](#url-display-handlers)
+  * [Favicons](#favicons)
+  * [Sidebar display](#sidebar-display)
+  * [Rendered scheme override](#rendered-scheme-override)
+  * [Custom in-page display](#custom-in-page-display)
+  * [Miscellaneous display customization](miscellaneous-display-customization)
+
+<!-- tocstop -->
+
+Prerequisites
+-------------
+
+Allowing to link external websites as
+either [other databases](https://musicbrainz.org/doc/Other_Databases_Relationship_Type/Whitelist) (for persistence issues)
+or [lyrics websites](https://musicbrainz.org/doc/Style/Relationships/URLs/Lyrics_whitelist) (for license issues)
+must be approved by a STYLE decision beforehand.
+See <https://musicbrainz.org/doc/Proposals>
+for how to request such STYLE decision,
+before even thinking about coding handlers.
+
+Other relationship types don’t require any STYLE decision,
+but having a self-assigned MBS ticket is still required.
+
+The following research work is usually expected about external websites to be handled:
+
+* Form and number of its URLs in the MusicBrainz database already;
+  - Tip: For a first approximate count, you can query the MusicBrainz API about URLs with (approximate) patterns;
+  - For example [`url:*/openlibrary.org/*`](https://musicbrainz.org/ws/2/url?query=url:*/openlibrary.org/*&limit=1&fmt=json) counts Open Library URLs (mainly).
+
+* Mapping between patterns of its URLs and MusicBrainz entities with relationship type(s);
+  - For example `/authors/` Open Library URLs match MusicBrainz artists.
+
+* Potential alternative forms of its URLs;
+  - For example `youtu.be/REF` is an alias for `www.youtube.com/watch?v=REF`.
+
+* Potential query (`?...`) and fragment (`#...`);
+  - For example `&list=...` is removed from YouTube video URLs taken from a playlist.
+
+* Potential patterns to be blocked (search pages, unreliable URLs...);
+  - For example `youtube.com/playlist` URLs are unwanted and blocked with a message.
+
+* Rationale and real examples for all the above.
+
+Expected behavior and rationale should be given in the ticket,
+and implementation details in the pull request and commit messages.
+
+It isn’t required to make a fine-grained implementation.
+Sometimes it is even impossible because of URL patterns.
+
+URL editing handlers
+--------------------
+
+Most of the handling of URLs, including all three of (URL) cleanup,
+(relationship type) autoselect and (URL relationship) validation, happen on
+[`root/static/scripts/edit/URLCleanup.js`](root/static/scripts/edit/URLCleanup.js).
+(Note: The name of this file is historical and will be changed in the future;
+See [MBS-12836](https://tickets.metabrainz.org/browse/MBS-12836).)
```

Comment:
One issue at a time.
The reason we didn’t already do is mentioned in the description of the ticket MBS-12836: “it will conflict with a lot of ongoing work.” And there is a huge pile of work in progress here, not even counting the changes I did not push as PR yet. In terms of refactoring, we already have the entity type category naming in progress.

Reply:
Ok, but then what's the point of saying this in the file? We can change the file when we change the rest and it shouldn't confuse anyone :/

Not a huge issue either way, just not sure of the usefulness of it.

Reply:
It is just documenting a known issue in the code naming convention, and this file is about coding conventions among other things.

As it seems to be too much information, I removed it in commit https://github.com/metabrainz/musicbrainz-server/pull/2395/commits/fcde0caea430705db2d4b1957b26295baf420752.
The ticket can be found indirectly anyway since there is also a link to the Jira component “URL cleanup” in the introduction.